openex-api/src/main/java/io/openex/config/AppSecurityConfig.java

```diff
@@ -26,165 +29,398 @@
+    User userLogin = null;
+    try {
+      userLogin = userManagement(emailAttribute, registrationId, rolesFromUser,
+          user.getFirstAttribute(
+              env.getProperty("openex.provider." + registrationId + ".firstname_attribute_key", String.class, "")),
+          user.getFirstAttribute(
+              env.getProperty("openex.provider." + registrationId + ".lastname_attribute_key", String.class, ""))
+      );
+    } catch (Exception e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
     }
 
-    @Bean
-    public OAuth2UserService<OAuth2UserRequest, OAuth2User> oauth2UserService() {
-        DefaultOAuth2UserService delegate = new DefaultOAuth2UserService();
-        return request -> oauthUserManagement(request.getAccessToken(), request.getClientRegistration(), delegate.loadUser(request));
+    if (userLogin != null) {
+      return userLogin;
     }
+
+    Saml2Error authError = new Saml2Error("invalid_token", "User conversion fail");
+    throw new Saml2AuthenticationException(authError);
+  }
+
+  public OidcUser oidcUserManagement(OAuth2AccessToken accessToken, ClientRegistration clientRegistration,
+      OAuth2User user) {
+    User loginUser = userOauth2Management(accessToken, clientRegistration, user);
+    return new OpenExOidcUser() {
+      @Override
+      public String getId() {
+        return loginUser.getId();
+      }
+
+      @Override
+      public boolean isAdmin() {
+        return loginUser.isAdmin();
+      }
+
+      @Override
+      public Map<String, Object> getClaims() {
+        return getAttributes();
+      }
+
+      @Override
+      public OidcUserInfo getUserInfo() {
+        return OidcUserInfo.builder().name(getName()).email(getEmail()).build();
+      }
+
+      @Override
+      public OidcIdToken getIdToken() {
+        return null;
+      }
+
+      @Override
+      public Map<String, Object> getAttributes() {
+        HashMap<String, Object> attributes = new HashMap<>();
+        attributes.put("id", loginUser.getId());
+        attributes.put("name", loginUser.getFirstname() + " " + loginUser.getLastname());
+        attributes.put("email", loginUser.getEmail());
+        return attributes;
+      }
+
+      @Override
+      public Collection<? extends GrantedAuthority> getAuthorities() {
+        List<SimpleGrantedAuthority> roles = new ArrayList<>();
+        roles.add(new SimpleGrantedAuthority(ROLE_USER));
+        if (loginUser.isAdmin()) {
+          roles.add(new SimpleGrantedAuthority(ROLE_ADMIN));
+        }
+        return roles;
+      }
+
+      @Override
+      public String getName() {
+        return loginUser.getFirstname() + " " + loginUser.getLastname();
+      }
+
+    };
+  }
+
+  public OAuth2User oAuth2UserManagement(OAuth2AccessToken accessToken, ClientRegistration clientRegistration,
+      OAuth2User user) {
+    User loginUser = userOauth2Management(accessToken, clientRegistration, user);
+    return new OpenExOAuth2User() {
```

Comment:
OpenExOAuth2User can be a class to move all this next line and be more readable
What do you think ?